### ISSUE TITLE
Clearly inform merchant (in UI/woo admin) when their bank account is not accepting deposits or has an error

### ISSUE THREAD

**Issue by haszari** [Automattic/woocommerce-payments#6294]
### Describe the bug
If deposits (payouts) to the merchant's bank account fail, we don't surface consistently surface this in the UI (though there should be email notifications when this happens). 

[Stripe exposes this state in the `external_account_bank_accounts` API, as `status=errored`.](https://stripe.com/docs/api/external_account_bank_accounts/object#account_bank_account_object-status)

The goal of this issue is to inform merchants in this state that their payout failed via a notice in WCPay dashboard.

This could be a regular notice, a task list item, a notice on the deposits card, or shown on the WCPay overview, e.g. as a high-priority notice in the top `Welcome` card.

### To Reproduce
* Visit the Stripe Express dashboard for your test WooPayments account (Payments → Overview → Account Details → Edit Details)
	* Update the payout bank account to a [failing test account](https://docs.stripe.com/connect/testing#payouts), e.g. `110000000`/`000111111116 `
	* Note your Stripe account's ID, e.g. `acct_123` (can be found in the browser devtools network response for `https://api.stripe.com/v1/account`)
* Authenticate via the [Stripe Dashboard](https://dashboard.stripe.com/) using your A8C creds.
* Use the [Stripe Shell](https://docs.stripe.com/connect/manual-payouts?shell=true&api=true&resource=accounts&action=retrieve) to create a manual deposit.
   ```sh
   stripe payouts create --stripe-account="acct_123" --amount="123" --currency="USD" 
   ```
* Use the [Stripe Shell](https://docs.stripe.com/connect/manual-payouts?shell=true&api=true&resource=accounts&action=retrieve) to check if the account's external account status is now `errored`.
   ```sh
   stripe accounts retrieve --stripe-account acct_123
   
   # Bank account's can be found in `external_accounts.data`
   ```
* Navigate back to your account's Payments Overview and notice there are no indications of bank account errors.

### Expected behavior
Show a notice to merchant saying something like:

> Your deposits are currently paused because a recent deposit failed. Please update your bank account details. 

Also show this info (payouts are interrupted and bank account needs fixing) everywhere it's relevant:

- Settings > Payments > Deposits
- Deposits list screen
- Anywhere else relevant

Related internal discussion (CSAT review): pchH0V-1cq-p2

**Comment by haszari** [comment]
Marking this as low priority as it's a relatively rare edge case. However for affected merchants it could be confusing and we could help more (by surfacing the error to UI).

**Comment by haszari** [comment]
Next steps:

- Confirm how to reproduce this scenario for testing. 
- Design iterations on wording and placement for the notice. 

Could add a notice to the `Deposits` card in the next deposit section, e.g. something like this: 

<img width="852" alt="Screenshot 2023-05-15 at 3 29 54 PM" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/5e852ce9-fdb5-46f1-8e36-a2f69c4c886a">

**Comment by souravdebnath1986** [comment]
The error for a payout is surfaced by Stripe in the payout object [here](https://stripe.com/docs/api/payouts/object#payout_object-failure_code). The list of payout errors is also shown. Stripe docs for [reference](https://stripe.com/docs/payouts#payout-failures)

**Comment by Jinksi** [comment]
Note for future explorers of this issue.

I've discovered that forcing a failed Stripe payout via the Stripe dashboard manual payout UI does **not** update the [external bank account](https://stripe.com/docs/api/external_account_bank_accounts/object#account_bank_account_object-status) `status` to `errored`.


<img src="https://github.com/Automattic/woocommerce-payments/assets/3147296/ad283638-eec8-4ebe-b036-41300f60e9f8" width="300" />


The payout object, however, **does** include failure status and details:


```json
"failure_code": "invalid_currency",
"failure_message": "The bank was unable to process this transfer because of its currency. This is probably because the bank account cannot accept payments in that currency.",
"status": "failed",
```

**Comment by Jinksi** [comment]
⭐ However!

The external bank account will show `status: errored` if the manual deposit is initiated via the [Stripe CLI/shell](https://stripe.com/docs/connect/manual-payouts?shell=true&api=true&resource=payouts&action=create) when the account already has a [failing test bank account](https://stripe.com/docs/connect/testing#payouts).

```sh
stripe payouts create --stripe-account="acct_123" --amount="10" --currency="USD" 

# creates a payout object with "status": "pending"
```


```sh
stripe external_accounts list acct_123

# Now the external account object will have errored status
# "status": "errored"

# also available via
stripe accounts retrieve --stripe-account acct_123
```

**Comment by haszari** [comment]
> can get the bank account into `status: errored` by initiating a manual deposit via the Stripe CLI when the account already has a [thing]

Nice! This might be a useful addition to our testing/dev cheat sheet 😁

**Comment by Jinksi** [comment]
@haszari as I was writing out testing instructions that I've added to this issue description, I noticed that there is a big obvious task list item that would resolve this issue.

This occurs when using the failing test account `110000000 | 000333333335 | Payout fails with a debit_not_authorized code.`

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/0a9d27f2-5a9a-4ab1-8908-399e50b4d0aa)

**Comment by Jinksi** [comment]
OK, so using the failing test account `110000000 | 000111111116 | Payout fails with a no_account code.` ~~will not show a task list item like the test above ☝️~~

**Update**: It did show after a delay.

@haszari If this task list item does reliably appear in the event of a deposit failure, do you think we should:
1. show another notice within the Deposits card specifically based on the bank account `errored` state, or
2. rely on the business details task list item to resolve this issue?

**Comment by Jinksi** [comment]
> Nice! This might be a useful addition to our testing/dev cheat sheet 😁

Added PdjTHR-2XG-p2

**Comment by haszari** [comment]
Thanks for digging in to this @Jinksi !

> there is a big obvious task list item that would resolve this issue.

That sounds good – and reduces the priority. 

Possibly we need to show a notice in the deposits overview, deposits screen etc as well. It's misleading/confusing if we show the deposits UI as normal when the bank account is missing or blocking deposits for whatever reason. Maybe we need to clarify the title … I'll do that. 

> If this task list item does reliably appear in the event of a deposit failure, do you think we should:
> show another notice within the Deposits card specifically based on the bank account errored state, or
> rely on the business details task list item to resolve this issue?

This is really a design/user experience question. Here's my take:

- If the merchant's deposits are failing, or bank account is missing
- We should alert merchant about this in the UI, near any info about deposits
  - primarily in the deposits card in overview screen
  - but maybe also in payment settings
  - and deposits list

@elizaan36 - keen for some design input on this maintenance issue, no particular urgency here (low occurrence/impact).

---

I actually re-experienced this issue last week and was going to log it! I have a screenshot which sums it up. When my bank account settings were broken, I got an email, but I couldn't tell from looking at WCPay UI (possibly I did have a notice in task list that I missed – I was focused on payments overview & settings screen at the time). With a critical config issue like this, we should probably communicate in more places (e.g. email only + task list might not be enough).

<img width="1551" alt="Email reminder to update bank account, but no in-app reminder or info" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/8aee6aa5-6961-4b9f-867c-52bb1a8a6f77">


Related, when a payout fails, it has a red icon and `Failed`, however no error message, no clues as to why it failed or if it's a one-off or if the account info is missing. We could log a separate issue for this maybe.

<img width="1399" alt="no error message when payout fails" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/636b6d74-8669-45aa-a413-c98108076844">

**Comment by haszari** [comment]
> Related, when a payout fails, it has a red icon and Failed, however no error message, no clues as to why it failed or if it's a one-off or if the account info is missing. We could log a separate issue for this maybe.

Logged!

- #8203

**Comment by elizaan36** [comment]
Thanks for catching this and it's great that we have all the juicy error messages from Stripe to provide the merchant. I can have a look at surfacing the errors (I'm thinking either an inline notice or tooltip) in the relevant places. 

Question - Does the "Update business details" task also appear on Woo Home?

**Comment by haszari** [comment]
Discussed this issue with @souravdebnath1986  in backlog meeting and we're keen to prioritise it. It looks like there is some design happening in parallel, hoping we can start prototyping/implementing technically. 

Answering some questions earlier in the thread…

> If this task list item does reliably appear in the event of a deposit failure, do you think we should:

> - show another notice within the Deposits card specifically based on the bank account errored state, or
> - rely on the business details task list item to resolve this issue?

I think we should show notices or make the "your bank account is not working" info visible anywhere related to deposits. i.e. redundantly communicate this important error.

> Question - Does the "Update business details" task also appear on Woo Home?

I don't know … @Jinksi can you comment? 

Although this message hints at the problem, it's maybe too corporate/vague. That notice looks more focused on "not yet fully onboarded" scenario; the scenario we're concerned about is "bank account is not accepting deposits and merchant cannot get payouts".

**Comment by Jinksi** [comment]
> Question - Does the "Update business details" task also appear on Woo Home?

Tested and confirmed that the "Update business details" tasks appears on **Payments → Overview** and **Woo Home**.

**Payments → Overview**
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/fc41308a-4f26-4b67-9ad6-7e98ea54d9e2)

**Woo Home**
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/2eb5e773-2d8d-4101-bba0-f1fbb289c4cf)

**Comment by jessy-p** [comment]
@Jinksi  Looking into the code, it seems that the task list items are coming from `restricted_soon` status, which seems to be independent of the external bank account [API](https://docs.stripe.com/api/external_account_bank_accounts/object#account_bank_account_object-status) that Stripe is  referring - pchH0V-1cq-p2#comment-1394 

When I update payout details on my (old, working) account to one of the bank accounts for payout failure, I do not get the tasks since my account as such is not restricted, but the external bank account API does give `errored` status. Could you check?

**Comment by haszari** [comment]
Note I've added some scope to the issue description, based on discussions so far. The goal of this issue is to surface any errors with bank accounts consistently, wherever appropriate in the UI. Each notice/UI can be handled as a separate issue or PR – i.e. incremental delivery. 

I just want to clarify that the issue is that we need to surface this error everywhere relevant, not just ensure we have a notice in one specific screen / task list. 

Here's the scope I added: 

---

Also show this info (payouts are interrupted and bank account needs fixing) everywhere it's relevant:

- Settings > Payments > Deposits
- Deposits list screen

**Comment by Jinksi** [comment]
> independent of the external bank account API

@jessy-p Yes I see the same behaviour – an `errored` bank account does not always result in a `restricted_soon` status. This highlights the need for a consistent notice for failing bank accounts, independent of the "Update business details" notice 💯

**Comment by elizaan36** [comment]
Got it, thanks for looking into this. I'll take a closer look at how to surface this error.

**Comment by jessy-p** [comment]
@elizaan36  Could you please have a look at the screenshots in 
- https://github.com/Automattic/woocommerce-payments/pull/8321 (Payments > Overview > Deposits )
- https://github.com/Automattic/woocommerce-payments/pull/8347 (Payments > Deposits)

and let me know if they are OK to merge?

**Comment by elizaan36** [comment]
Hey @jessy-p @haszari I think the notices in the PRs look great from a design perspective. I added comments about conditionally hiding the less urgent notice though.

I would have shared the same solution so I think you nailed it. If this is still a problem for merchants once these messages are implemented, we can consider using the task list as an additional failsafe.
